### PR TITLE
Feature/using supertypescript

### PR DESCRIPTION
**Pull request by sshankar19**
Using the latest in supertype.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,12 +2,11 @@ FROM node:8.3-alpine
 
 EXPOSE 3001
 RUN apk update && apk add --no-cache bash
+RUN apk add git
```
Comment (hackerrdave) [inline Dockerfile]: I would add this to the previous line:
```
RUN apk update && apk add --no-cache bash git
```

File: lib/typescript.js [hunk 2]
```diff
@@ -52,7 +67,12 @@ function bindDecorators (objectTemplate) {
      * @returns {unknown} unknown.
      */
     this.remote = function remote(defineProperty) {
-        return objectTemplate.remote(defineProperty, objectTemplate);
+        if (objectTemplate.remote) {
```
Comment (hackerrdave) [inline lib/typescript.js:70]: why are these if conditions needed now?

File: Dockerfile (hunk 1 above)
Comment (sshankar19) [reply Dockerfile]: I actually don't need this. I wasn't supposed to commit this, thanks for catching it

File: lib/typescript.js (hunk 2 above)
Comment (sshankar19) [reply lib/typescript.js:70]: I changed all of the decorator exports in Supertype to be seperate from the other functionality (since they really shouldn't be sharing state with ObjectTemplate or Supertype, they're just typescript decorators)

Since everything isn't all in one object anymore (and perhaps due to some context change as well), `objectTemplate.remote`, `objectTemplate.supertypeClass`, `objectTemplate.property`, etc. may be undefined in an app that's only using supertype in persistor. 

_However_, in a semotus app, the decorator fields (`remote`, `property`, `supertypeClass`) get added onto the `RemoteObjectTemplate` (base object template), and that's the proper path we want to take (this path will eventually take us back to the supertype decorators, but that's a different story).

So, in short, it's just a check to see if we've loaded semotus.

Comment (hackerrdave) [reply lib/typescript.js:70]: ahh perfect thanks, makes sense

**Review by hackerrdave — APPROVED** [review]

**Review by srksag-zz — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -6,8 +6,6 @@ RUN apk update && apk add --no-cache bash
...
 
 COPY ./wait-for-it.sh .
-
-COPY ./package.json .
-RUN yarn install --production=false
-
-COPY ./ ./
\ No newline at end of file
+COPY ./ ./
+RUN npm install --production=false
```
Comment (nikmash) [inline Dockerfile]: doesn't this command by default do the same thing as `--production=false`?

Comment (nikmash) [reply Dockerfile]: @sshankar19 just checked this is not even a valid option: https://docs.npmjs.com/cli/install the presence of `--production` will be a production `npm install` otherwise it will be a non-production install (bring in devDependencies). Using a boolean is not a valid option

Source: https://docs.npmjs.com/cli/install

**Review by nsaenz7 — APPROVED** [review]

**Review by nikmash — APPROVED** [review]
Looks good!
